src/katsdpcontroller/product_controller.py

```diff
@@ -1510,14 +1518,23 @@ async def dsim_signals(self, dsim: str, signals_str: str, period: Optional[int]
                 break
         else:
             raise FailReply(f"No dsim named {dsim}")
+        if dsim_task.katcp_connection is None:
+            raise FailReply(f"No katcp connection to {dsim}")
```

Comment:
I've always meant to check
* How does this scenario transpire?
* Why don't we induce/create our own connection?

Reply:
If the dsim dies we'll set the `katcp_connection` to None.